### PR TITLE
Add support for "scripts-aliases" in composer.json

### PR DESCRIPTION
**Pull request by TravisCarden**
Closes: https://github.com/composer/composer/issues/11420

I added the functionality and tests and updated relevant documentation I found.

### CODE REVIEW

File: src/Composer/Command/ScriptAliasCommand.php [hunk 1]
```diff
@@ -27,11 +27,23 @@ class ScriptAliasCommand extends BaseCommand
...
 
-    public function __construct(string $script, ?string $description)
+    /**
+     * @param string[]|null $aliases
+     */
+    public function __construct(string $script, ?string $description, ?array $aliases = [])
     {
         $this->script = $script;
         $this->description = $description ?? 'Runs the '.$script.' script as defined in composer.json';
+        $this->aliases = $aliases ?? [];
```
Comment (Seldaek) [inline src/Composer/Command/ScriptAliasCommand.php:40]:
```suggestion
        $this->aliases = $aliases;
```

**Comment by Seldaek** [comment]
Nice, looks good to me overall, solid complete PR with all the things updated 👍🏻

**Comment by TravisCarden** [comment]
Great! Back to you, @Seldaek.

**Comment by Seldaek** [comment]
Thanks, merging now that I branched out 2.6

**Comment by ArnaudLigny** [comment]
Hello @Seldaek, I already use this new feature (throught the snapshot) locally and would like use it on prod, but I can't because my CI checks the validtity of the composer file (and `scripts-aliases` is not valide with Composer 2.6.6).

Do you think you can create a new release taking this feature into account? Thanks in advance.

**Comment by Seldaek** [comment]
https://github.com/composer/composer/releases/tag/2.7.0 is out now

**Pull request merged**